15314f1ddb882700dc9fabc5ca961943/update/sys_script_include_b630a09ddb9020d01206fd641d9619df.xml

```diff
@@ -718,20 +718,355 @@ IllumioUtils.prototype = {
+	 * @param {String} key for the newly created label dimension
+	 * @param {String} pceConfig sys_id of the configuration record
+	 * @returns {Boolean} status of whether the label dimension has to be created
+	 */
+	validateLabelDimensionCreation: function (key, pceConfigSysId) {
+		try {
+			var matchFound = false;
+			
+			var pceConfig = new GlideRecord("x_illu2_illumio_illumio_pce_conf");
+			if(pceConfig.get(pceConfigSysId)) {
+				var responseDict = this.getLabelDimensions(pceConfig);
+				if (!responseDict["status"]) {
+					return matchFound;
+				}
+				response = responseDict["response"];
+				response_status = responseDict["statusCode"];
+
+				if (response_status >= 200 && response_status < 300) {
+					response = JSON.parse(response.getBody());
+
+					// Iterate over each dimension and intialize record, if not already present, and update it.
+					for (var i = 0; i < response.length; i++) {
+						this.createOrUpdateLabelDimensionRecord(response[i], pceConfig);
+
+						if (response[i]["key"] == key) {
+							matchFound = true;
+							gs.debug("A label dimension with key " + key + " is already present on Illumio PCE for PCE configuration " + pceConfig.getValue("name") + ". Replacing the label dimension details with the ones obtained from Illumio PCE.");
+						}
+					}
+				} else {
+					gs.error("Error while validating the creation of label dimension with key - " + key + " for the PCE configuration " + pceConfig.getValue("name") + ". Status code - " + response_status + ". Response - " + JSON.stringify(response));
+					return false;
+				}
+			}
+		} catch (exc) {
+			gs.error("Error while validating the creation of label dimension with key - " + key + ". Error - " + exc);
+			return false;
+		}
+		return matchFound;
+	},
+	
+	/**
+	 * Retry failed label dimension creation
+	 * When the label dimensions creation fails on PCE, this method is to be called with the next scheduled script execution while fetching the label dimensions.
+	 * This method would retry creating the label dimensions aaded on ServiceNow onto the coorelating Illumio PCE tenant.
+	 */
+	retryFailedLabelDimensionCreation: function () {
+		var labelDimensionGr = new GlideRecord("x_illu2_illumio_illumio_label_dimension");
+		labelDimensionGr.addNullQuery("href");
+		labelDimensionGr.query();
+		while (labelDimensionGr.next()) {
+			try {
+				var details = {
+					"key": labelDimensionGr.getValue("key"),
+					"display_name_singular": labelDimensionGr.getValue("display_name_singular"),
+					"display_name_plural": labelDimensionGr.getValue("display_name_plural"),
+					"label_type_initial": labelDimensionGr.getValue("label_type_initial"),
+					"illumio_pce_configuration": labelDimensionGr.getValue("illumio_pce_configuration")
+				};
+				
+				gs.debug("Trying to create label dimension " + labelDimensionGr.getValue("key") + " on Illumio PCE for PCE configuration " + labelDimensionGr.getValue("illumio_pce_configuration") + ".");
```

Comment:
nit: Its better to replace `labelDimensionGr.getValue("key")` with `details['key']` and labelDimensionGr.getValue("illumio_pce_configuration") with `details['illumio_pce_configuration']` since details refers to the record that is in scope of iteration.

Reply:
Noted! I’ll update this with branch - “n-arbitrary-label-dimensions”